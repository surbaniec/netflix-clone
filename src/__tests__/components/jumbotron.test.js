import { render } from '@testing-library/react';
import Jumbotron from '../../components/jumbotron';
import jumboData from '../../fixtures/jumbo.json';

describe('<Jumbotron />', () => {
  it('renders the <Jumbtron /> with populated data', () => {
    const { container, getByText, getByAltText, getByTestId } = render(
      <Jumbotron.Container>
        {jumboData.map((item) => (
          <Jumbotron key={item.id}>
            <Jumbotron.Pane>
              <Jumbotron.Title>{item.title}</Jumbotron.Title>
              <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
            </Jumbotron.Pane>
            <Jumbotron.Pane>
              <Jumbotron.Image
                src={item.image}
                alt={item.alt}
                data-testid={`${item.id}-jumbo-image`}
              />
            </Jumbotron.Pane>
          </Jumbotron>
        ))}
      </Jumbotron.Container>
    );

    expect(getByText('Enjoy on your TV.')).toBeTruthy();
    expect(getByAltText('Tiger King on Netflix')).toBeTruthy();
    expect(getByTestId('1-jumbo-image')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
