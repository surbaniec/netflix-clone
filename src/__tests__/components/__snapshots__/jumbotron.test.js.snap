// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders the <Jumbtron /> with populated data 1`] = `
<div
  class="jumbotron__Container-sc-1al52go-6 fcDFtY"
>
  <div
    class="jumbotron__Item-sc-1al52go-0 bCfsnK"
  >
    <div
      class="jumbotron__Inner-sc-1al52go-1 dsliNL"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-1al52go-2 hcnpUw"
      >
        <h1
          class="jumbotron__Title-sc-1al52go-3 hvCzJz"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-1al52go-4 gOTSYS"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-1al52go-2 hcnpUw"
      >
        <img
          alt="Tiger King on Netflix"
          class="jumbotron__Image-sc-1al52go-5 klRzZU"
          data-testid="1-jumbo-image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-1al52go-0 bCfsnK"
  >
    <div
      class="jumbotron__Inner-sc-1al52go-1 dsliNL"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-1al52go-2 hcnpUw"
      >
        <h1
          class="jumbotron__Title-sc-1al52go-3 hvCzJz"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-1al52go-4 gOTSYS"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-1al52go-2 hcnpUw"
      >
        <img
          alt="Watch on mobile"
          class="jumbotron__Image-sc-1al52go-5 klRzZU"
          data-testid="2-jumbo-image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-1al52go-0 bCfsnK"
  >
    <div
      class="jumbotron__Inner-sc-1al52go-1 dsliNL"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-1al52go-2 hcnpUw"
      >
        <h1
          class="jumbotron__Title-sc-1al52go-3 hvCzJz"
        >
          Watch everywhere.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-1al52go-4 gOTSYS"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-1al52go-2 hcnpUw"
      >
        <img
          alt="Money Heist on Netflix"
          class="jumbotron__Image-sc-1al52go-5 klRzZU"
          data-testid="3-jumbo-image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
